### PR TITLE
Support both POST and GET

### PR DESCRIPTION
**Pull request by Rikuoja**
Implement #64

### CODE REVIEW

**Comment by Rikuoja** [comment]
Will not merge yet, will also add `multipart/form-data` support for POSTing binary files.

File: tools/network.py [hunk 1]
```diff
@@ -83,7 +154,39 @@ def fetch_raw(
...
         request_blocking.setAuthCfg(authcfg_id)
-    _ = request_blocking.get(req)
+    # QNetworkRequest *only* supports get and post. No idea why.
+    if method == "get":
+        _ = request_blocking.get(req)
+    elif method == "post":
+        if data:
+            # Support JSON
+            byte_data = bytes(json.dumps(data), encoding)
+            req.setRawHeader(b"Content-Type", bytes("application/json", encoding))
```
Comment (LKajan) [inline tools/network.py:164]: The used encoding should be added to the header like `application/json; charset=utf-8`

File: tools/network.py [hunk 2]
```diff
@@ -83,7 +154,39 @@ def fetch_raw(
...
+            # https://github.com/requests/toolbelt/blob/master/requests_toolbelt/multipart/encoder.py
+            boundary = uuid4().hex
+            byte_boundary = bytes(f"\r\n--{boundary}\r\n", encoding)
+            last_byte_boundary = bytes(f"\r\n--{boundary}--\r\n", encoding)
+            byte_boundary_with_disposition = (
+                byte_boundary
+                + bytes(CONTENT_DISPOSITION_FORM_DATA, encoding)
+                + bytes(CONTENT_TYPE_FORM_DATA, encoding)
+            )
+            byte_data = (
```
Comment (LKajan) [inline tools/network.py:176]: Can you please revisit this code part?
I think now there is an extra `byte_boundary_with_disposition` at the start. Should just be `byte_boundary`
And all the Content-Disposition: form-data input names, filenames and content types are now static in the `byte_boundary_with_disposition` regardless of the file.

File: tools/network.py [hunk 3]
```diff
@@ -32,6 +34,10 @@
 CONTENT_DISPOSITION_BYTE_HEADER = QByteArray(
     bytes(CONTENT_DISPOSITION_HEADER, ENCODING)
 )
+CONTENT_DISPOSITION_FORM_DATA = (
```
Comment (LKajan) [inline tools/network.py:37]: I think these should not be static. See the comment below.

**Review by LKajan — CHANGES_REQUESTED** [review]
I think there are some issues in the multipart file upload that should be solved. Now the receiver sees all the files as file.xml.
Otherwise looks good.

File: tools/network.py (hunk 2 above)
Comment (LKajan) [reply tools/network.py:176]: Ah, I got confused how the join method works. The first point of mine is not relevant. Sorry for that.

File: tools/network.py [hunk 4]
```diff
@@ -68,6 +95,50 @@ def fetch_raw(
...
+    :param data: Dictionary to send in the request body
+    :param files: Files to send multipart-encoded
+    :return: bytes of the content and default name of the file or empty string
+    """
+    return request_raw(url, "post", encoding, authcfg_id, None, data, files)
+
+
+def request_raw(
+    url: str,
+    method: str = "get",
```
Comment (LKajan) [inline tools/network.py:123]: This could be changed to Enum. Or at least Type should be `Literal["get", "post"]` to make clear with supported values.

**Review by LKajan — APPROVED** [review]

**Pull request merged**